src/toil_scripts/gatk_germline/hard_filter.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python2.7
+from __future__ import print_function
+import os
+
+from bd2k.util.humanize import human2bytes
+from toil.job import PromisedRequirement
+
+from toil_scripts.lib.files import upload_or_move_job
+from toil_scripts.tools.variant_filters import gatk_genotype_gvcf, gatk_select_variants, \
+    gatk_variant_filtration
+
+
+def hard_filter_pipeline(job, uuid, gvcf_id, config):
+    """
+    Runs GATK Hard Filtering on HaplotypeCaller output. Writes separate SNP and INDEL
+    VCF files to output directory.
+
+    0: GenotypeGVCFs            0 --> 1 --> 3 --> 5
+    1: Select SNPs                |
+    2: Select INDELs              +-> 2 --> 4 --> 5
+    3: Apply SNP Filter
+    4: Apply INDEL Filter
+    5: Write filtered VCFs to output directory
+
+    :param job: Toil Job instance
+    :param str uuid: Unique sample identifier
+    :param str gvcf_id: GVCF FileStoreID
+    :param Namespace config: Pipeline configuration options and shared files
+    :return: SNP and INDEL FileStoreIDs
+    :rtype: tuple
+    """
+    job.fileStore.logToMaster('Running Hard Filter on {}'.format(uuid))
+
+    genotype_gvcf_disk = PromisedRequirement(lambda x: 2*x.size + human2bytes('5G'), gvcf_id)
+    genotype_gvcf = job.wrapJobFn(gatk_genotype_gvcf,
+                                  dict(uuid=gvcf_id),
+                                  config,
+                                  memory=config.xmx, disk=genotype_gvcf_disk)
+
+    select_snps_disk = PromisedRequirement(lambda x: 2*x.size + human2bytes('5G'),
+                                           genotype_gvcf.rv())
+    select_snps = job.wrapJobFn(gatk_select_variants,
+                                'SNP',
+                                genotype_gvcf.rv(),
+                                config,
+                                disk=select_snps_disk)
+
+    snp_filter_disk = PromisedRequirement(lambda x: 2*x.size + human2bytes('5G'),
```

Comment:
Where ever a constant value (e.g., the '8G' in this line) comes in, you need to document where this constant comes from.